icu4c/source/common/charstr.cpp

```diff
@@ -141,6 +141,39 @@ CharString &CharString::append(const char *s, int32_t sLength, UErrorCode &error
     return *this;
 }
 
+CharString &CharString::appendNumber(int32_t number, UErrorCode &status) {
```

Comment:
Can you add a unit tests for this function please?  The call site only ever formats numbers between 1 and 15.  I want to make sure the logic is correct for larger numbers.  You can add the test wherever CharString is tested.